security/pkg/k8s/configutil_test.go

```diff
@@ -306,3 +306,71 @@ func createFakeLister(kubeClient *fake.Clientset) informersv1.ConfigMapInformer
 	kube.WaitForCacheSync(ctx.Done(), configmapInformer.HasSynced)
 	return informerFactory.Core().V1().ConfigMaps()
 }
+
+func Test_insertData(t *testing.T) {
+	type args struct {
+		cm   *v1.ConfigMap
+		data map[string]string
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "unchanged",
+			args: args{
+				cm:   createConfigMap(namespaceName, configMapName, map[string]string{"foo": "bar"}),
+				data: nil,
+			},
+			want: false,
+		},
+		{
+			name: "unchanged",
+			args: args{
+				cm:   createConfigMap(namespaceName, configMapName, map[string]string{"foo": "bar"}),
+				data: map[string]string{"foo": "bar"},
+			},
+			want: false,
+		},
+		{
+			name: "changed",
+			args: args{
+				cm:   createConfigMap(namespaceName, configMapName, map[string]string{"foo": "bar"}),
+				data: map[string]string{"bar": "foo"},
+			},
+			want: true,
+		},
+		{
+			name: "changed",
+			args: args{
+				cm:   createConfigMap(namespaceName, configMapName, map[string]string{"foo": "bar"}),
+				data: map[string]string{"foo": "foo"},
+			},
+			want: true,
+		},
+		{
+			name: "changed",
+			args: args{
+				cm:   createConfigMap(namespaceName, configMapName, nil),
+				data: map[string]string{"bar": "foo"},
+			},
+			want: true,
+		},
+		{
+			name: "changed",
+			args: args{
+				cm:   createConfigMap(namespaceName, configMapName, nil),
+				data: nil,
+			},
+			want: true,
```

Comment:
This should be checking the resulting map as well. I would encourage you to focus on getting the most coverage on tests - not just lines of code coverage, but really testing the functionality.

Reply:
@howardjohn  
Fixed.
Understood, thanks.